Managers/UInputMan.cpp

```diff
@@ -228,176 +228,179 @@ int UInputMan::InputScheme::Create()
-        m_aInputMapping[INPUT_L_DOWN].SetDirection(0, 1, JOYDIR_TWO);
-        m_aInputMapping[INPUT_L_LEFT].SetDirection(0, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_L_RIGHT].SetDirection(0, 0, JOYDIR_TWO);
-        m_aInputMapping[INPUT_R_UP].SetDirection(2, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_R_DOWN].SetDirection(2, 0, JOYDIR_TWO);
-        m_aInputMapping[INPUT_R_LEFT].SetDirection(1, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_R_RIGHT].SetDirection(1, 0, JOYDIR_TWO);
-        m_aInputMapping[INPUT_FIRE].SetDirection(0, 2, JOYDIR_ONE);
-        m_aInputMapping[INPUT_JUMP].SetDirection(0, 1, JOYDIR_ONE);
-        m_aInputMapping[INPUT_CROUCH].SetDirection(0, 1, JOYDIR_TWO);
-        m_aInputMapping[INPUT_PIEMENU].SetDirection(0, 2, JOYDIR_TWO);
-
-
-        m_aInputMapping[INPUT_WEAPON_CHANGE_PREV].SetDirection(3, 0, JOYDIR_ONE);
-        m_aInputMapping[INPUT_WEAPON_CHANGE_NEXT].SetDirection(3, 0, JOYDIR_TWO);
-		m_aInputMapping[INPUT_WEAPON_RELOAD].SetDirection(3, 1, JOYDIR_ONE);
-        m_aInputMapping[INPUT_WEAPON_PICKUP].SetDirection(3, 1, JOYDIR_TWO);
-    }
+		case PRESET_P3DEFAULT:
+			m_aInputMapping[INPUT_L_UP].SetKey(KEY_W);
+			m_aInputMapping[INPUT_L_DOWN].SetKey(KEY_S);
+			m_aInputMapping[INPUT_L_LEFT].SetKey(KEY_A);
+			m_aInputMapping[INPUT_L_RIGHT].SetKey(KEY_D);
+			m_aInputMapping[INPUT_R_UP].SetKey(KEY_W);
+			m_aInputMapping[INPUT_R_DOWN].SetKey(KEY_S);
+			m_aInputMapping[INPUT_R_LEFT].SetKey(KEY_A);
+			m_aInputMapping[INPUT_R_RIGHT].SetKey(KEY_D);
+			m_aInputMapping[INPUT_FIRE].SetKey(KEY_H);
+			m_aInputMapping[INPUT_AIM].SetKey(KEY_J);
+			m_aInputMapping[INPUT_AIM_UP].SetKey(KEY_W);
+			m_aInputMapping[INPUT_AIM_DOWN].SetKey(KEY_S);
+			m_aInputMapping[INPUT_PIEMENU].SetKey(KEY_K);
+			m_aInputMapping[INPUT_JUMP].SetKey(KEY_L);
+			m_aInputMapping[INPUT_CROUCH].SetKey(KEY_STOP);
+			m_aInputMapping[INPUT_NEXT].SetKey(KEY_U);
+			m_aInputMapping[INPUT_PREV].SetKey(KEY_Y);
+			break;
+
+		case PRESET_P4DEFAULT:
+			m_aInputMapping[INPUT_L_UP].SetKey(KEY_UP);
+			m_aInputMapping[INPUT_L_DOWN].SetKey(KEY_DOWN);
+			m_aInputMapping[INPUT_L_LEFT].SetKey(KEY_LEFT);
+			m_aInputMapping[INPUT_L_RIGHT].SetKey(KEY_RIGHT);
+			m_aInputMapping[INPUT_R_UP].SetKey(KEY_UP);
+			m_aInputMapping[INPUT_R_DOWN].SetKey(KEY_DOWN);
+			m_aInputMapping[INPUT_R_LEFT].SetKey(KEY_LEFT);
+			m_aInputMapping[INPUT_R_RIGHT].SetKey(KEY_RIGHT);
+			m_aInputMapping[INPUT_FIRE].SetKey(KEY_1_PAD);
+			m_aInputMapping[INPUT_AIM].SetKey(KEY_2_PAD);
+			m_aInputMapping[INPUT_AIM_UP].SetKey(KEY_UP);
+			m_aInputMapping[INPUT_AIM_DOWN].SetKey(KEY_DOWN);
+			m_aInputMapping[INPUT_PIEMENU].SetKey(KEY_3_PAD);
+			m_aInputMapping[INPUT_JUMP].SetKey(KEY_DEL_PAD);
+			m_aInputMapping[INPUT_CROUCH].SetKey(KEY_STOP);
+			m_aInputMapping[INPUT_NEXT].SetKey(KEY_5_PAD);
+			m_aInputMapping[INPUT_PREV].SetKey(KEY_4_PAD);
+			break;
+
+			// Some generic defaults; no real preset is set
+		default:
```

Comment:
Sorry if you guys have already been over this, or I'm missing something, but when is this default case used/useful? Shouldn't this always be called with a valid preset number, so the default case should abort out?

Reply:
after discussion in discord, stays as is.